Ginger/GingerCore/ALM/OctaneCore.cs

```diff
@@ -226,5 +365,520 @@ public override bool IsServerConnected()
+                                }
+                            }
+                            else
+                            {
+                                //#param exist
+                                if (isflowControlParam == true)
+                                {
+                                    if (!(stepActivityVar is VariableSelectionList))
+                                    {
+                                        //flow control param must be Selection List so transform it
+                                        stepActivity.Variables.Remove(stepActivityVar);
+                                        stepActivityVar = new VariableSelectionList();
+                                        stepActivityVar.Name = param;
+                                        stepActivity.AddVariable(stepActivityVar);
+                                        stepActivity.AutomationStatus = eActivityAutomationStatus.Development;//reset status because flow control param was added
+                                    }
+                                }
+                                else if (isflowControlParam == false)
+                                {
+                                    if (stepActivityVar is VariableSelectionList)
+                                    {
+                                        //change it to be string variable
+                                        stepActivity.Variables.Remove(stepActivityVar);
+                                        stepActivityVar = new VariableString();
+                                        stepActivityVar.Name = param;
+                                        ((VariableString)stepActivityVar).InitialStringValue = paramSelectedValue;
+                                        stepActivity.AddVariable(stepActivityVar);
+                                        stepActivity.AutomationStatus = eActivityAutomationStatus.Development;//reset status because flow control param was removed
+                                    }
+                                }
+                            }
+
+                            //add the variable selected value                          
+                            if (stepActivityVar is VariableSelectionList)
+                            {
+                                OptionalValue stepActivityVarOptionalVar = ((VariableSelectionList)stepActivityVar).OptionalValuesList.Where(x => x.Value == paramSelectedValue).FirstOrDefault();
+                                if (stepActivityVarOptionalVar == null)
+                                {
+                                    //no such variable value option so add it
+                                    stepActivityVarOptionalVar = new OptionalValue(paramSelectedValue);
+                                    ((VariableSelectionList)stepActivityVar).OptionalValuesList.Add(stepActivityVarOptionalVar);
+                                    if (isflowControlParam == true)
```

Comment:
Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6317725)